dap-python.el

```diff
@@ -70,6 +70,112 @@ as the pyenv version then also return nil. This works around https://github.com/
           executable))
     (executable-find command)))
 
+(cl-defstruct dap-python--point
+  (line (:type integer) :named t)
+  (character (:type integer) :named t))
+
+(cl-defstruct dap-python--location
+  (start (:type dap-python--point) :named t)
+  (end (:type dap-python--point) :named t))
+
+(cl-defstruct dap-python--symbol
+  (name (:type string) :named t)
+  (type (:type string) :named t)
+  (location (:type dap-python--location) :named t))
+
+(cl-defgeneric dap-python--equal (lhs rhs)
+  (:documentation "Check if lhs and rhs are equal"))
+
+(cl-defmethod dap-python--equal ((lhs symbol) (rhs symbol))
+  (eq lhs rhs))
+
+(cl-defmethod dap-python--equal ((lhs integer) (rhs integer))
+  (eq lhs rhs))
+
+(cl-defmethod dap-python--equal ((lhs string) (rhs string))
+  (string-equal lhs rhs))
+
+(cl-defmethod dap-python--equal ((lhs list) (rhs list))
+  (and (dap-python--equal (length lhs) (length rhs))
+       (-reduce (lambda (x y) (and x y)) (-zip-with 'dap-python--equal lhs rhs))))
+
+(cl-defmethod dap-python--equal ((lhs dap-python--point) (rhs dap-python--point))
+  (and (dap-python--equal (dap-python--point-line lhs) (dap-python--point-line rhs))
```

Comment:
I think `equal` will just work here, it compares lisp structure, so all of your custom comparison should be covered with it.